data-science-onramp/feature-engineering/citibike.ipynb

```diff
@@ -0,0 +1,385 @@
+    "\n",
+    "+ `BLOB_PREFIX`: indicates the folder where the files are stored.\n",
+    "+ `DIR_NAME`: holds the name of the local folder where the files will be downloaded to."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BUCKET_NAME = 'your-bucket-name'\n",
+    "BLOB_PREFIX = 'clean_citibike_data.csv.gz/part'\n",
+    "DIR_NAME = 'citibike_data'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load the data\n",
+    "\n",
+    "Run the following command to create a local folder where the dataset files will be stored."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir $DIR_NAME"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell will retrieve all of the files with the `BLOB_PREFIX` defined above and download them into the folder. You will also create a list of the file names so they can be referenced later when loading the data into a dataframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create storage client\n",
+    "storage_client = storage.Client()\n",
+    "\n",
+    "# List files in the bucket with the specified prefix\n",
+    "blobs = storage_client.list_blobs(BUCKET_NAME, prefix=BLOB_PREFIX)\n",
+    "\n",
+    "# Go through the files and save them into the local folder\n",
+    "filenames = []\n",
+    "for i, blob in enumerate(blobs):\n",
+    "    filename = f'{DIR_NAME}/citibike{i}.csv.gz'\n",
+    "    blob.download_to_filename(filename)\n",
+    "    filenames.append(filename)\n",
+    "    print('Downloaded file: ' + str(blob.name))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, you can load the files into a dataframe. \n",
+    "\n",
+    "First, define the schema. From this dataset, you will need 4 columns:\n",
+    "\n",
+    "+ Start time: extract the day of the week and date of when the trip is started\n",
+    "+ Stop time: extract the day of the week and date of when the trip has ended\n",
+    "+ Start station: find out how many trips started at a station\n",
+    "+ End station: find out how many trips ended at a station"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "COLUMNS = (\n",
+    "    'starttime',\n",
+    "    'stoptime',\n",
+    "    'start_station_id',\n",
+    "    'end_station_id',\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, run the following cell to loop through the files downloaded to the local folder, create a `Pandas DataFrame`, and view the first 10 rows."
```

Comment:
Nit - number 10 should be spelled out (https://developers.google.com/style/numbers)